### PR TITLE
feat: add LogicalInterface model

### PR DESCRIPTION
**Pull request by djothi**
**Description:**

Add LogicalInterface model.

### CODE REVIEW

File: netbox_cmdb/netbox_cmdb/models/interface.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    )
+    mode = models.CharField(
+        choices=LOGICAL_INTERFACE_MODE_CHOICES,
+        blank=True,
+        null=True,
+        default=None,
+        max_length=20,
+        help_text="Interface mode (802.1Q)",
+    )
+    untagged_vlan = models.ForeignKey(
```
Comment (kpetremann) [inline netbox_cmdb/netbox_cmdb/models/interface.py:98]: it is missing a constraint as untagged is only "access" ports and tagged/native are only for "trunk" ports.

authorized settings only:
- untagged
- tagged
- tagged + native

forbidden settings:
- untagged + tagged
- untagged + native
- untagged + tagged + native

File: netbox_cmdb/netbox_cmdb/models/interface.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+class LogicalInterface(ChangeLoggedModel):
+    """A logical interface configuration."""
+
+    index = models.PositiveSmallIntegerField()
+    parent_interface = models.ForeignKey(
+        to="DeviceInterface",
+        related_name="%(class)s",
+        on_delete=models.CASCADE,
+        blank=True,
```
Comment (fdomain) [inline netbox_cmdb/netbox_cmdb/models/interface.py:64]: In which cases do we allow to have an empty parent_interface? Loopback interfaces ?

Comment (fdomain) [reply netbox_cmdb/netbox_cmdb/models/interface.py:64]: as there is no name field for the logical interface, it seems to be mandatory to have a parent_interface

**Review by fdomain — CHANGES_REQUESTED** [review]

File: netbox_cmdb/netbox_cmdb/models/interface.py [hunk 3]
```diff
@@ -52,3 +64,95 @@ def __str__(self):
...
+
+    index = models.PositiveSmallIntegerField()
+    enabled = models.BooleanField(default=True)
+    state = models.CharField(
+        max_length=50,
+        choices=AssetStateChoices,
+        default=AssetStateChoices.STATE_STAGING,
+        help_text="State of this DeviceInterface",
+    )
+    monitoring_state = models.CharField(
```
[the remark this answers is not in the document]
Comment (djothi) [reply netbox_cmdb/netbox_cmdb/models/interface.py:80]: IIRC we briefly talked about it, but nothing more. Let's schedule a dedicated meeting. The solution could also impact `BGPSession` monitoring fields

[the remark this answers is not in the document]
Comment (kpetremann) [reply netbox_cmdb/netbox_cmdb/models/interface.py:80]: it has already been discussed indeed, there was a page and a consensus :)
I'll share the page in slack

**Review by kpetremann — APPROVED** [review]

**Review by fdomain — APPROVED** [review]

**Pull request merged**